src/Raven.Server/Smuggler/Documents/StreamDestination.cs

```diff
@@ -1557,19 +1575,27 @@ private async ValueTask WriteUniqueAttachmentStreamsAsync(Document document, Smu
                     }
 
                     progress.Attachments.ReadCount++;
-
-                    if (_attachmentStreamsAlreadyExported.Add(hash))
+                    if (attachment.TryGet(nameof(AttachmentName.Flags), out AttachmentFlags flags) == false || flags == AttachmentFlags.None)
                     {
-                        await using (var stream = _source.GetAttachmentStream(hash, out string tag))
+                        if (_attachmentStreamsAlreadyExported.Add(hash))
                         {
-                            if (stream == null)
+                            await using (var stream = _source.GetAttachmentStream(hash, out string tag))
                             {
-                                progress.Attachments.ErroredCount++;
-                                throw new ArgumentException($"Document {document.Id} seems to have an attachment hash: {hash}, but no correlating hash was found in the storage.");
+                                if (stream == null)
+                                {
+                                    progress.Attachments.ErroredCount++;
+                                    throw new ArgumentException($"Document {document.Id} seems to have an attachment hash: {hash}, but no correlating hash was found in the storage.");
+                                }
+                                await WriteAttachmentStreamAsync(hash, stream, tag);
                             }
-                            await WriteAttachmentStreamAsync(hash, stream, tag);
                         }
                     }
+                    else if(flags == AttachmentFlags.Retired)
+                    {
+                        
```

Comment:
TODO

Reply:
well I think we need to do something like
`progress.Attachments.RetiredCount++`

here